### PR TITLE
LF: check viewtype is a monomorphic record

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs [hunk 1]
```diff
@@ -865,7 +865,12 @@ checkDefTypeSyn DefTypeSyn{synParams,synType} = do
 -- | Check that an interface definition is well defined.
 checkIface :: MonadGamma m => Module -> DefInterface -> m ()
 checkIface m iface = do
-  let tcon = Qualified PRSelf (moduleName m) (intName iface)
+  
+  -- check view
+  tycon <- match _TCon (EExpectedViewType viewtype) viewtype
+  DefDataType _loc _naem _serializable tparams dataCons <- inWorld (lookupDataType tycon)
```
Comment (akrmn) [inline compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:871]:
```suggestion
  DefDataType _loc _naem serializable tparams dataCons <- inWorld (lookupDataType tycon)
```

I think we also want to check that `serializable == IsSerializable True`

Comment (akrmn) [reply compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:871]: that should happen in `compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Serializability.hs`, though

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs [hunk 2]
```diff
@@ -385,6 +386,8 @@ instance Pretty Error where
       "tried to perform key lookup or fetch on template " <> pretty tpl
     EExpectedOptionalType typ -> do
       "expected list type, but found: " <> pretty typ
+    EExpectedViewType typ -> do
+      "expected monomorphic record type in view type, but found:" <> pretty typ
```
Comment (akrmn) [inline compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs:390]: I'm not sure "monomorphic" is exactly the thing we want, after all, `Foo Bool` is monomorphic for this `Foo`:

```haskell
data Foo a = Foo
  { bar : Text
  , baz : a
  }
```

I would say something like this instead, 

```suggestion
      "expected serializable record type without type arguments in view type, but found:" <> pretty typ  
```

I'm not super happy about "without type arguments", but "of kind *" feels too technical.

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs (hunk 1 above)
Comment (akrmn) [reply compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:871]: sorry, this is already checked in the module I linked x.x

Comment (remyhaemmerle-da) [reply compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs:871]: It is already in `compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Serializability.hs`

File: compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs (hunk 2 above)
Comment (akrmn) [reply compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs:390]: Yes I just noticed that one, I think it's fine to be consistent then.
I also noticed some of these (existing) errors are slightly different between the scala and haskell typecheckers, but that's more of a gardening thing.

**Review by akrmn — APPROVED** [review]
Thank you!
The point about `monomorphic` vs alternative wording can be explored later, for now consistency is better.

**Pull request merged**